pkg/restore/restic_restore_action.go

```diff
@@ -71,10 +75,23 @@ func (a *ResticRestoreAction) Execute(input *velero.RestoreItemActionExecuteInpu
 
 	log := a.logger.WithField("pod", kube.NamespaceAndName(&pod))
 
-	volumeSnapshots := restic.GetPodSnapshotAnnotations(&pod)
+	opts := restic.NewPodVolumeBackupListOptions(input.Restore.Spec.BackupName)
+	podVolumeBackupList, err := a.podVolumeBackupClient.List(opts)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	var podVolumeBackups []*velerov1api.PodVolumeBackup
+	for i, pvb := range podVolumeBackupList.Items {
+		podVolumeBackups[i] = &pvb
+	}
+	volumeSnapshots := restic.GetVolumesForPod(podVolumeBackups, &pod)
 	if len(volumeSnapshots) == 0 {
-		log.Debug("No restic snapshot ID annotations found")
-		return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
+		volumeSnapshots = restic.GetPodSnapshotAnnotations(&pod)
+		if len(volumeSnapshots) == 0 {
+			log.Debug("No restic snapshot ID annotations found")
+			return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
```

Comment:
Do we still want to return early here? If there are PVBs created with the new code, len(volumeSnapshots) would always be 0, even if there are PVB objects returned by the client.

Reply:
we only hit this code if there are no PVBs per L89. I think this will be clearer if the "first check PVBs, then check annotations if no PVBs" logic moves into the helper func.